src/webots/gui/WbMainWindow.cpp

```diff
@@ -1737,9 +1723,9 @@ void WbMainWindow::uploadFinished() {
   } else {
     WbLog::info(tr("link: %1\n").arg(url));
 
-    mLinkWindow = new WbLinkWindow(this);
-    mLinkWindow->setLabelLink(url);
-    mLinkWindow->exec();
+    WbLinkWindow *linkWindow = new WbLinkWindow(this);
```

Comment:
I am afraid there is a memory leak here... It seems this object should get deleted when we delete the WbMainWindow object, but we should probably delete it earlier when this linkWindow is not needed any more. Is it the case?

Reply:
It should be fixed by not creating it using the `new` syntax